### PR TITLE
[backport 2.7] Fix possible close_notify/ClientHello confusion

### PR DESCRIPTION
**Pull request by mpg**
This is the 2.7 backport of #3050

It also has an extra commit to align some timeout values with the ones used in the 2.16 branch.

### CODE REVIEW

File: tests/ssl-opt.sh [hunk 1]
```diff
@@ -5308,8 +5308,8 @@ run_test    "DTLS proxy: 3d, min handshake, resumption" \
             -p "$P_PXY drop=5 delay=5 duplicate=5" \
             "$P_SRV dtls=1 hs_timeout=250-10000 tickets=0 auth_mode=none \
              psk=abc123 debug_level=3" \
-            "$P_CLI dtls=1 hs_timeout=250-10000 tickets=0 psk=abc123 \
-             debug_level=3 reconnect=1 read_timeout=1000 max_resend=10 \
+            "$P_CLI dtls=1 hs_timeout=500-10000 tickets=0 psk=abc123 \
```
Comment (gilles-peskine-arm) [inline tests/ssl-opt.sh]: You've changed this timeout (and likewise for one other test case below). I'm ok with the change since it aligns 2.7 with the other stable branches. However please do this in a separate commit.

**Comment by mpg** [comment]
@gilles-peskine-arm Thanks for catching the unintended and inconsistent change. I now aligned the timeout with the 2.16 branch for that entire series of tests, in a separate commit for clarity. Please review again.
